### PR TITLE
Fixed errno 150 issue

### PR DESCRIPTION
**Pull request by stillalivx**
Hey! I fixed my issue (#106). I just added a conditional to check if primary key field is type integer, in that case, it will add the not nullable property to the field "belongsTo"

### CODE REVIEW

File: lib/helpers/fields.ts [hunk 1]
```diff
@@ -18,8 +18,12 @@ export function addFieldToSchema(
 
       const relationshipPKType: FieldTypeString = fieldOptions.type.relationship.model
         .getComputedPrimaryType();
-      
-      table[relationshipPKType](fieldOptions.name);
+
+      if (relationshipPKType === "integer") {
+        table[relationshipPKType](fieldOptions.name).notNullable();
```
Comment (eveningkid) [inline lib/helpers/fields.ts:23]: Quick question: how does using `notNullable()` solves it? I thought the issue was between `signed/unsigned int` from #106. Could you please explain it more thoroughly? I am afraid I don't understand..!

**Review by eveningkid — COMMENTED** [review]
Hi,

Thank you for sending this PR, I really appreciate it!

There's just one question on my side before we merge, thanks.

Keep in touch

File: lib/helpers/fields.ts (hunk 1 above)
Comment (stillalivx) [reply lib/helpers/fields.ts:23]: Sorry for the late response, I was checking the issue and I found more problems but I resolved them.
First, you are right, the issue was about the signed/unsigned int. That happens when somebody adds on the primaryKey field the ```autoIncrement``` property.
Also, in the field that contains the foreign key, automatically adds the allow null property, so we have to handle that. The primaryKey doesn't allow the null.
As I said, I already changed the code to fix those issues. Can you request the changes? 
Thanks!

**Comment by stillalivx** [comment]
Hello! Do you have some comments?

File: lib/model.ts [hunk 2]
```diff
@@ -157,6 +166,15 @@ export class Model {
         : field.type;
   }
 
+  /**/
```
Comment (eveningkid) [inline lib/model.ts:169]: We are a missing a description here of what it does. You can check other methods to see what's the comment style in use :)

File: lib/helpers/fields.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { FieldOptions, DataTypes, FieldTypeString } from "../data-types.ts";
+import {FieldOptions, DataTypes, FieldTypeString, FieldType, FieldProps} from "../data-types.ts";
```
Comment (eveningkid) [inline lib/helpers/fields.ts:1]: I don't know why we lost the spaces before and after braces. Can you just put them back please?

**Review by eveningkid — CHANGES_REQUESTED** [review]
Hi,

Sorry I only got back to you.

You did a great job, please see my two comments and we'll be good to merge.

It's just a matter of minutes I would say and once it's done I will merge it straight away.

Thank you for your help :)

**Comment by stillalivx** [comment]
That's it! The changes are done. :)

**Review by eveningkid — APPROVED** [review]
Great, thanks man! 😁

Have a good day

**Pull request merged**